provision.sh

```diff
@@ -14,31 +14,47 @@ echo "---------------------------------------------"
 echo "Running vagrant provisioning"
 echo "---------------------------------------------"
 
+# install heroku toolbelt
+echo "-------------- Installing heroku toolbelt -------------------------"
+#wget -O- https://toolbelt.heroku.com/install-ubuntu.sh | sh
+# These shell script snippets are directly taken from heroku installation script
+# We want to avoid the apt-get update
+# add heroku repository to apt
+echo "deb http://toolbelt.heroku.com/ubuntu ./" > /etc/apt/sources.list.d/heroku.list
+# install heroku's release key for package verification
+wget -O- https://toolbelt.heroku.com/apt/release.key | apt-key add -
+
 echo "------------- Updating packages and their dependencies -----------"
 apt-get update -y # no need for sudo, and -y is needed to bypass the yes-no
+
+# install the toolbelt
+apt-get install heroku-toolbelt --no-install-recommends -y
+
 echo "-------------- Installing build-essential and pip ----------------"
-apt-get install -y build-essential python-pip
+# install gcc and g++ and other build basics to ensure most software works
+# install man too
+apt-get install build-essential python-pip man -y --no-install-recommends
 
-# install heroku toolbelt
-echo "-------------- Installing heroku toolbelt -------------------------"
-wget -O- https://toolbelt.heroku.com/install-ubuntu.sh | sh
+# needed for heroku toolbelt
+# notice that this is not a rigorous Ruby install, where we typically use rvm
+apt-get install ruby --no-install-recommends -y
```

Comment:
Is this really needed? Cause I have not run into troubles without this. I can use Heroku normally without installing ruby. I thought it was already preinstalled on the base box.